catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/config/ConfigurationApplication.java

```diff
@@ -392,10 +393,11 @@ private void findDifferences(List<Map<String, Object>> innerList,
     }
 
     private void stopImageryEndpoints(List<String> imageryEndpointsToStop) {
-        for (String endpoint : imageryEndpointsToStop) {
+        for (Iterator<String> iterator = imageryEndpointsToStop.iterator(); iterator.hasNext(); ) {
+            String endpoint = iterator.next();
             try {
                 httpProxy.stop(endpoint);
-                imageryEndpoints.remove(endpoint);
+                iterator.remove();
```

Comment:
The `httpProxy.stop(endpoint)` shows up three separate times in this class, so it might be worth refactoring out to a method.

Then this entire method could be refactored to:
```java
imageryEndpointsToStop.forEach(this::tryStopHttpProxy);
imageryEndpoints.removeAll(imageryEndpointsToStop);
```